connector/openshift/openshift.go

```diff
@@ -115,9 +115,30 @@ func (c *Config) Open(id string, logger log.Logger) (conn connector.Connector, e
 		Scopes:      []string{"user:info"},
 		RedirectURL: c.RedirectURI,
 	}
+
+	if err := openshiftConnector.checkOAuth2EndpointsAvailability(ctx); err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to query oauth2 endpoint: %w",
+			err)
+	}
+
 	return &openshiftConnector, nil
 }
 
+func (c *openshiftConnector) checkOAuth2EndpointsAvailability(ctx context.Context) error {
+	for _, endpoint := range []string{c.oauth2Config.Endpoint.AuthURL, c.oauth2Config.Endpoint.TokenURL} {
+		req, err := http.NewRequest(http.MethodHead, endpoint, nil)
+		if err != nil {
+			return err
+		}
+		_, err = c.httpClient.Do(req.WithContext(ctx))
+		if err != nil {
+			return err
+		}
```

Comment:
Note, you are not actually checking the HTTP status code. Arguably, a 5xx or a 404 should be treated as an error; however, I'm not sure if the providers can be relied on to sufficient extent that they return the "right" status code (some HTTP server implementations might keep a separate route map per method and thus return a 404, when instead they should return a 405 Method Not Allowed). So probably no action required.